### PR TITLE
#135 Implement Java version check.

### PR DESCRIPTION
**Pull request by cbeach47**
##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]

##### Description of change

From issue #135 - Java version check implemented for all projects.  Currently Java 8 is required.

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

**Comment by bjagg** [comment]
Looks great! :+1:  ... What would it take to allow a range of versions? I'm thinking we will support Java 9 and 10 soon.

**Comment by cbeach47** [comment]
Once Java 9 / Java 10 are supported, it'd be fairly quick to support multiple values - not sure if a range would work since the JavaVersion.* are just enums, but at very least it can be enhanced to check a list/array of values.

File: build.gradle [hunk 1]
```diff
@@ -21,6 +21,13 @@ ext {
     isWindows = OperatingSystem.current().isWindows()
 }
 
+allprojects {
+    def targetJavaVersion = JavaVersion.VERSION_1_8;
+    def currentJavaVersion = JavaVersion.current();
+    if (currentJavaVersion != targetJavaVersion) 
+        throw new GradleException("Incorrect Java version; required is ${targetJavaVersion}, but found ${currentJavaVersion}");
```
Comment (drewwills) [inline build.gradle]: Would this check be better as a `task`?  As written, it fails in the Gradle initialization phase.

Something like...
```
task versionCheck {
    doLast {
        def supportedJavaVersions = [ JavaVersion.VERSION_1_8 ];
        def currentJavaVersion = JavaVersion.current();
        if (!supportedJavaVersions.contains(targetJavaVersion)) {
            throw new GradleException("Incorrect Java version; found ${currentJavaVersion} but . supported versions are:  ${supportedJavaVersions}");
        }
    }
}
```

Comment (drewwills) [reply build.gradle]: You then have to make specify that `portalProperties dependsOn versionCheck`.

Comment (cbeach47) [reply build.gradle]: Good point, I'll switch that around.

**Comment by cbeach47** [comment]
Supports a collection of versions, and I set it to be a dependency of compileJava, so it won't unnecessarily block other uPortal-start tasks.

**Pull request merged**